src/id_token_provider.erl

```diff
@@ -70,7 +72,9 @@ handle_info({refresh, Provider}, State) ->
   end,
   {noreply, State}.
 
-maybe_refresh(Provider) ->
+maybe_refresh(Provider, #{force_refresh := true}) ->
+  refresh(Provider);
+maybe_refresh(Provider, _Opts) ->
```

Comment:
not sure I understand this correctly, but will the second clause ever be called without dispatching to `refresh` in the end?

Reply:
I just updated id_token.erl, and now I am only setting `force_revalidate=true` if we get `{error, no_public_key_matches}`. I think that this might be the case if there are concurrent calls to `validate` which means we can have a scenario like this:

1. Thread A tries to validate an id_token, but realises that the keys have expired.
2. Thread B tries to validate an id_token, but realises that the keys have expired.
3. Now Thread A executes and refreshes keys and saves them in the local cache BEFORE thread B has managed to run any further.
4. Thread B now checks the local cache in `maybe_refresh` and realises that the new keys are no longer expired and instead of refreshing the keys again, it returns the keys in the cache. 

There is also a asynchronous refresh job in the background, see https://github.com/kivra/id_token/blob/284063ea7883de1ea566d573cbbfcb9191ce2109/src/id_token_provider.erl#L48, that adds more possibilities for concurrent updates.

Reply:
alright, makes sense! 👍